add docker build environment

d-schneider opened the pull request:
Updated to .NET Framework 4.6.2 and .NET Core 2.2.
Updated .csproj files to the new Format.
Added a Dockerfile for a windows build environment.

discostu105 commented on building/windowsdockerbuild.cmd:
```diff
@@ -0,0 +1,15 @@
+
+if exist %MSBUILD% (
+    dotnet restore ..\src
+    %MSBUILD% msbuild.targets /l:FileLogger,Microsoft.Build.Engine;logfile=msbuild.log /target:%target%
```
this build does not really use the docker image, does it?

d-schneider replied:
This file was only used in the docker container to build SuperDump.

I changed it in commit https://github.com/Dynatrace/superdump/pull/162/commits/2ae9708064b9c855674179e8634a44046339c134 to start the build in the docker container.

discostu105 commented on Dockerfile.WindowsBuild:
```diff
@@ -0,0 +1,63 @@
+	start-Process vs_buildtools.exe -ArgumentList '--quiet --norestart --nocache --add Microsoft.VisualStudio.Workload.MSBuildTools --add Microsoft.VisualStudio.Workload.NetCoreBuildTools' -Wait; \
+    Remove-Item -Force vs_buildtools.exe
+
+# default MS symbol servers. override this with your own list if necessary.
```
Is this really necessary for the build-image?

discostu105 commented:
Generally, I'd expect the following build files:
 * `build-windows-all.cmd` // regular windows build, non-docker
 * `build-docker-windows-build-image.cmd` // prepares a windows docker image with all necessary build tools
 * `build-windows-all-withdocker.cmd` // windows build that uses the docker-build-image. output in `/build` subdirectory
 * `build-docker-windows-superdumpservice.cmd` // creates a windows image that runs superdumpservice
 * `build-docker-linux-analysis.cmd` // creates linux image that can do linux coredump analysis

d-schneider commented:
I renamed the build scripts according to the feedback and added documentation.
Additionally, now the build of the Windows Service and Analyzer now starts automatically when running the docker container.

discostu105 commented on building/README.md:
```diff
@@ -0,0 +1,29 @@
+    * DebugDiag
+
+ * build-docker-linux-analysis.cmd
+   This script builds the Linux Analyzer and creates a docker container containing it. Docker has to be installed for this.
```
nit: "creates a docker image..."

discostu105 commented on building/README.md:
```diff
@@ -0,0 +1,29 @@
+   This creates a docker image that contains all requirements for building the Windows components.
+
+ * build-windows-all-with-docker.cmd
+   Builds the Windows comnponents using the image created by the previous script
```
typo: comnponents
